cli/src/pcluster3_config_converter/pcluster3_config_converter.py

```diff
@@ -0,0 +1,1000 @@
+                pcluster3_model[pcluster3_attribute] = attribute
+        except Exception as e:
+            _error(f"Wrong type for {pcluster2_attribute} in {pcluster2_model} section: {e}")
+
+    def write_configuration_file(self):
+        """Write converted yaml to the output file."""
+        if not os.path.isfile(self.output_file):
+            try:
+                config_folder = os.path.dirname(self.output_file) or "."
+                os.makedirs(config_folder)
+            except OSError as e:
+                if e.errno != errno.EEXIST:
+                    raise
+
+            # Fix permissions
+            with open(self.output_file, "a", encoding="utf-8"):
+                os.chmod(self.output_file, stat.S_IRUSR | stat.S_IWUSR)
+
+        # Write configuration to disk
+        if self.output_file:
+            with open(self.output_file, "w", encoding="utf-8") as config_file:
+                config_file.write(self.comments)
+                yaml.dump(
+                    self.pcluster3_configuration,
+                    config_file,
+                    sort_keys=True,
+                    Dumper=ConfigDumper,
+                    default_flow_style=False,
+                )
+
+    def validate_cluster_section_name(self):
+        """
+        Validate cluster section name.
+
+        If cluster_template is specified in the command, use specifeid cluster_template to find the cluster section.
+        If cluster_label is not specified, use cluster template specified in global section.
+        IF no cluster_label sepcified in the cluster section, verify if default cluster section exist.
+        """
+        if self.cluster_label:
+            self.cluster_section_name = f"cluster {self.cluster_label}"
+        elif self.config_parser.get("global", "cluster_template", fallback=None):
+            cluster_template = self.config_parser.get("global", "cluster_template", fallback=None)
+            self.cluster_section_name = f"cluster {cluster_template}"
+        elif "cluster default" in self.config_parser:
+            self.cluster_section_name = "cluster default"
+        else:
+            _error("Can not find a valid cluster section.")
+        if self.cluster_section_name not in self.config_parser.sections():
+            _error("The specified cluster section is not in the configuration.")
+
+    def validate_required_fields(self):
+        """Validate required fields for conversion."""
+        self.comments += (
+            "# Volume encrypted is default to True in ParallelCluster3 while it is default to False in "
+            "ParallelCluster2.\n"
+        )
+        _note(
+            "Volume encrypted is default to True in ParallelCluster3 while it is default to False in "
+            "ParallelCluster2."
+        )
+        if self.cluster_config_get("scheduler") == "slurm":
+            self.comments += (
+                "# Imds/Secured is default to True in ParallelCluster3 for Slurm while it is default to "
```

Comment:
I think it might be more common to have nested structures represented with dot-notation, so here `Imds.Secured` rather than `Imds/Secured` (esp considering something like CSS selectors or JmesPath)

Reply:
Changed